core/src/com/unciv/models/Spy.kt

```diff
@@ -151,22 +153,25 @@ class Spy private constructor() : IsPartOfGameInfoSerialization {
     }
 
     private fun startStealingTech() {
-        setAction(SpyAction.StealingTech)
         progressTowardsStealingTech = 0
-        turnsRemainingForAction = getTurnsRemainingToStealTech()
+        setAction(SpyAction.StealingTech)
+        endTurn()
```

Comment:
We want to check immediately if we can steal technology and maybe send a message.
In a very rare case where the technology cost is 0 we might even steal a technology on this turn.